bootloader/bootloadertest/bootloadertest.go

```diff
@@ -146,3 +163,106 @@ func (b *MockBootloader) SetRecoverySystemEnv(recoverySystemDir string, blVars m
 	b.RecoverySystemBootVars = blVars
 	return nil
 }
+
+// SetRunKernelImageEnabledKernel sets the current kernel "symlink" as returned
+// by Kernel(); returns' a restore function to set it back to what it was
+// before.
+func (b *MockBootloader) SetRunKernelImageEnabledKernel(kernel snap.PlaceInfo) func() {
+	old := b.runKernelImageEnabledKernel
+	b.runKernelImageEnabledKernel = kernel
+	return func() {
+		b.runKernelImageEnabledKernel = old
+	}
+}
+
+// SetRunKernelImageEnabledTryKernel sets the current try-kernel "symlink" as
+// returned by TryKernel(). If set to nil, TryKernel()'s second return value
+// will be false; returns' a restore function to set it back to what it was
+// before.
+func (b *MockBootloader) SetRunKernelImageEnabledTryKernel(kernel snap.PlaceInfo) func() {
+	old := b.runKernelImageEnabledTryKernel
+	b.runKernelImageEnabledTryKernel = kernel
+	return func() {
+		b.runKernelImageEnabledTryKernel = old
+	}
+}
+
+// SetRunKernelImageErrorFunction allows setting an error to be returned for the
+// specified function; it returns a restore function to set it back to what it
+// was before
+func (b *MockBootloader) SetRunKernelImageErrorFunction(f string, err error) func() {
```

Comment:
shouldn't the name be s/ErrorFunction/FunctionError/ ? otherwise it sounds to me we are setting a function which is not the case